// @logo-rn/logo-avatar/src/AvatarCss.ts
import { Dimensions, StyleSheet } from "react-native";
import { GHOST_COLOR, PRIMARY_LIGHT_COLOR } from "@logo-rn/shared-utils/utils/constants/CSS_CONSTANTS";

const styles = StyleSheet.create({
    avatar: {
        backgroundColor: PRIMARY_LIGHT_COLOR,
        alignItems: "center",
        justifyContent: "center"
    },
    image: {

    },
    backdrop: {
        backgroundColor: 'rgba(0, 0, 0, 0.5)',
    },
    modalCard: {
        display: "flex",
        position: "absolute",
        alignItems: "center",
        top: 200,
        left: -195,
        width: Dimensions.get('window').width,
        height: Dimensions.get('window').height,
        backgroundColor: GHOST_COLOR,
        paddingVertical: 16
    },
    rowContainer: {
        display: "flex",
        flexDirection: "row",
    },
    modalButton: {
        width: 80,
        height: 80,
        marginHorizontal: "15%",
        marginTop: 24
    },
    modalText: {
        fontWeight: "500",
        fontFamily: "Inter",
        fontSize: 16,
    },
    avatarText: {
        fontWeight: "600",
        fontFamily: "Inter",
        fontSize: 18,
        color: GHOST_COLOR
    }

});

export default styles;
